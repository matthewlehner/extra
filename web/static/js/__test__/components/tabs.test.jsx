import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Tabs from "components/tabs";

const panels = [{
  label: "Monday",
  content: "hi bud"
}, {
  label: "Tuesday",
  content: "great day"
}];

describe("Tabs", () => {
  it("renders tabs", () => {
    const props = { panels };
    const tree = renderer.create(
      <MemoryRouter>
        <Tabs {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it("can specify active panel", () => {
    const props = {
      activePanel: panels[1].label,
      panels
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Tabs {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  describe("interactions", () => {
    it("switches panes", () => {
      const props = { panels };
      const component = renderer.create(
        <MemoryRouter>
          <Tabs {...props} />
        </MemoryRouter>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      tree.children[0].children[1].props.onClick();

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
