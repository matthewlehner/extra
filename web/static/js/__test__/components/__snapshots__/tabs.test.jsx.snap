// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs can specify active panel 1`] = `
<div
  className="tab-container"
>
  <nav
    className="tablist"
    role="tablist"
  >
    <a
      aria-controls="monday"
      aria-selected={false}
      className="tab"
      data-title="Monday"
      href="#monday"
      id="monday-tab"
      role="tab"
    >
      Monday
    </a>
    <a
      aria-controls="tuesday"
      aria-selected={true}
      className="tab"
      data-title="Tuesday"
      href="#tuesday"
      id="tuesday-tab"
      role="tab"
    >
      Tuesday
    </a>
  </nav>
  <div
    aria-hidden={true}
    aria-labelledby="monday-tab"
    className="tabpanel"
    id="monday"
    role="tabpanel"
  >
    hi bud
  </div>
  <div
    aria-hidden={false}
    aria-labelledby="tuesday-tab"
    className="tabpanel"
    id="tuesday"
    role="tabpanel"
  >
    great day
  </div>
</div>
`;

exports[`Tabs renders tabs 1`] = `
<div
  className="tab-container"
>
  <nav
    className="tablist"
    role="tablist"
  >
    <a
      aria-controls="monday"
      aria-selected={true}
      className="tab"
      data-title="Monday"
      href="#monday"
      id="monday-tab"
      role="tab"
    >
      Monday
    </a>
    <a
      aria-controls="tuesday"
      aria-selected={false}
      className="tab"
      data-title="Tuesday"
      href="#tuesday"
      id="tuesday-tab"
      role="tab"
    >
      Tuesday
    </a>
  </nav>
  <div
    aria-hidden={false}
    aria-labelledby="monday-tab"
    className="tabpanel"
    id="monday"
    role="tabpanel"
  >
    hi bud
  </div>
  <div
    aria-hidden={true}
    aria-labelledby="tuesday-tab"
    className="tabpanel"
    id="tuesday"
    role="tabpanel"
  >
    great day
  </div>
</div>
`;
